packages/api/blocks-api/src/blocks/factories/createOneOfBlock.ts

```diff
@@ -319,5 +327,9 @@ export function createOneOfBlock<BlockMap extends BaseBlockMap>(
         migrate = nameOrOptions.migrate;
     }
 
-    return createBlock(OneOfBlockData, OneOfBlockInput, { name, blockMeta: new Meta(OneOfBlockData), migrate });
+    return createBlock(OneOfBlockData, OneOfBlockInput, {
+        name,
+        blockMeta: new Meta(OneOfBlockData),
+        migrate,
+    }) as unknown as OneOfBlock<BlockMap, BlockData, BlockInput>;
```

Comment:
I've added the as cast to suppress the following error:

```
Type 'Block<OneOfBlockDataInterface<BlockMap, BlockDataInterface>, OneOfBlockInputInterface<BlockMap, SimpleBlockInputInterface>>' is not assignable to type 'OneOfBlock<BlockMap, BlockData, BlockInput>'.
  Types of property 'blockDataFactory' are incompatible.
    Type 'BlockDataFactory<OneOfBlockDataInterface<BlockMap, BlockDataInterface>>' is not assignable to type 'BlockDataFactory<OneOfBlockDataInterface<BlockMap, BlockData>>'.
      Type 'OneOfBlockDataInterface<BlockMap, BlockDataInterface>' is not assignable to type 'OneOfBlockDataInterface<BlockMap, BlockData>'.
        Type 'OneOfBlockDataInterface<BlockMap, BlockDataInterface>' is not assignable to type 'BlockData'.
          'OneOfBlockDataInterface<BlockMap, BlockDataInterface>' is assignable to the constraint of type 'BlockData', but 'BlockData' could be instantiated with a different subtype of constraint 'BlockDataInterface'.ts(2322)
```

I don't think it's possible to fix this without adding such a cast somewhere, so I decided to invest little effort and don't type options, internal classes etc. correctly. Do you think this okay?

Reply:
I'm not aware enough of how hard it is to have proper typing here, due to my lack of knowledge of the code-base, but I would say this should really be avoided :sweat_smile:

Reply:
What do you mean? Avoid the as cast?

Reply:
Yes

Reply:
I'm not sure if this can be achieved when using generics though

Reply:
I just checked out the code for it and I'm not familiar with the code here either but I agree with Obi. Even if I don't like the cast, I haven't come up with anything better here, if that's even possible. I could only imagine that Niko will come up with something else here.

Reply:
@Reaw I've invested some more time and moved the as casts further into the factory: 25f396f997417047fe0b052917cd1c52946a8d3b. I don't know how the remaining errors could be fixed, and I don't want to invest more time into it, since it's perfectly fine to have as casts in the library. Feel free to give it a shot though.